lib/utils/search.js

```diff
@@ -0,0 +1,58 @@
+function getRowNodeFromNode(node) {
+  while (node && !('rowIndex' in node)) {
+    node = node.parentNode;
+  }
+
+  return node;
+}
+
+function findNextFrom(terminal, searchTerm, selection = {}, options = {}) {
+  const {startRow} = selection;
+  let {startOffset} = selection;
+  const rowCount = terminal.getRowCount();
+
+  let foundIndex = -1;
+  let foundRow;
+  for (let i = 0; i <= rowCount; i++) {
+    // increment/decrement foundRow while ensuring it's always 0 <= foundRow < rowCount
+    foundRow = (startRow + (rowCount * 2) + (options.direction * i)) % rowCount;
+    const rowText = terminal.getRowText(foundRow);
+    foundIndex = rowText[options.direction === 1 ? 'indexOf' : 'lastIndexOf'](searchTerm, startOffset);
+    if (foundIndex !== -1 && foundIndex !== startOffset) {
+      break;
+    }
+    startOffset = undefined;
+  }
+
+  if (foundIndex === -1) {
+    return null;
+  }
+
+  return {
+    index: foundIndex,
+    row: foundRow
+  };
+}
+
+// find a searchTerm, highlighting it (and scrolling to it if currently off screen)
```

Comment:
This comment needs to be updated now (you moved the scrolling away from this function.)